apps/cowswap-frontend/src/modules/ordersTable/pure/OrdersTableContainer/utils/getOrderParams.ts

```diff
@@ -38,18 +38,14 @@ export function getOrderParams(
   const balance = balances[order.inputToken.address]?.value
   const allowance = allowances[order.inputToken.address]?.value
 
-  let hasEnoughBalance, hasEnoughAllowance
-
-  if (order.partiallyFillable) {
-    // When balance or allowance are undefined (loading state), show as true
-    // When loaded, check there's at least PERCENTAGE_FOR_PARTIAL_FILLS of balance/allowance to consider it as enough
-    const amount = sellAmount.multiply(PERCENTAGE_FOR_PARTIAL_FILLS)
-    hasEnoughBalance = balance === undefined || isEnoughAmount(amount, balance)
-    hasEnoughAllowance = allowance === undefined || isEnoughAmount(amount, allowance)
-  } else {
-    hasEnoughBalance = isEnoughAmount(sellAmount, balance)
-    hasEnoughAllowance = isEnoughAmount(sellAmount, allowance)
-  }
+  const [hasEnoughBalance, hasEnoughAllowance] = (() => {
+    // Check there's at least PERCENTAGE_FOR_PARTIAL_FILLS of balance/allowance to consider it as enough
+    const amount = order.partiallyFillable ? sellAmount.multiply(PERCENTAGE_FOR_PARTIAL_FILLS) : sellAmount
+    const hasEnoughBalance = isEnoughAmount(amount, balance)
+    const hasEnoughAllowance = isEnoughAmount(amount, allowance)
+
+    return [hasEnoughBalance, hasEnoughAllowance]
+  })()
```

Comment:
While this is cleaner, I am not a fan of this pattern.

Can you create an external fn for this instead?

Reply:
Depending on the case, i find it fine. My rule is, if this makes the body to big, i think local aux functions are best, but if not, IIFE might be a good option.

There's also a cost on context switching to go to a private function, and having to check the declared types and result.

Anyways, here i moved it. I see your point